src/components/main.js

```diff
@@ -1,6 +1,19 @@
-import React from 'react'
+import React from "react";
 
 export const Main = props => {
-  const { email } = props.history.location.state
-  return <div>Hello {email}</div>
-}
+    let welcomeMsg;
+
+    if (props.history) {
+        if (props.history.location.state) {
+            const { email } = props.history.location.state;
+            welcomeMsg = `Hello ${email}`
+        } else {
+            welcomeMsg = "Please log in or create a new account"
+        }
+    }
```

Comment:
The above could be simplified as follow: 
```suggestion
    let welcomeMsg = "Please log in or create a new account"; 
    
    if (props.history && props.history.location.state) {
      const { email } = props.history.location.state;
      welcomeMsg = `Hello ${email}`;
    }
```

Reply:
Reworked as suggested